### PR TITLE
[KOGITO-4303] Modify existing task API to update task

### PR DESCRIPTION
**Pull request by fjtirado**
Adding new put method for modify model without changing phase
Adding new /phases resource

### CODE REVIEW

File: integration-tests/integration-tests-springboot/src/it/integration-tests-springboot-it/src/test/java/org/kie/kogito/integrationtests/springboot/TaskTest.java [hunk 1]
```diff
@@ -49,4 +53,44 @@ void testJsonSchema() {
...
+                .post("/approvals")
+                .then()
+                .statusCode(201)
+                .extract()
+                .path("id");
+
+        String taskId = given()
+                .contentType(ContentType.JSON)
+                .when()
+                .get("/approvals/" + processId + "/tasks?user=admin&group=managers")
```
Comment (cristianonicolai) [inline integration-tests/integration-tests-springboot/src/it/integration-tests-springboot-it/src/test/java/org/kie/kogito/integrationtests/springboot/TaskTest.java:74]: you can use path parameters directly, see https://github.com/rest-assured/rest-assured/wiki/Usage#path-parameters

File: integration-tests/integration-tests-springboot/src/it/integration-tests-springboot-it/src/test/java/org/kie/kogito/integrationtests/springboot/TaskTest.java [hunk 2]
```diff
@@ -49,4 +53,44 @@ void testJsonSchema() {
...
+                .then()
+                .statusCode(200)
+                .extract()
+                .path("[0].id");
+
+        given().contentType(ContentType.JSON)
+                .when()
+                .body(Collections.singletonMap("approved", true))
+                .patch(String.format("/approvals/%s/firstLineApproval/%s?user=admin&group=managers", processId, taskId))
```
Comment (cristianonicolai) [inline integration-tests/integration-tests-springboot/src/it/integration-tests-springboot-it/src/test/java/org/kie/kogito/integrationtests/springboot/TaskTest.java:83]: same here regarding path param

File: integration-tests/integration-tests-springboot/src/it/integration-tests-springboot-it/src/test/java/org/kie/kogito/integrationtests/springboot/TaskTest.java [hunk 3]
```diff
@@ -49,4 +53,44 @@ void testJsonSchema() {
...
+                .body("approved", is(true));
+
+        given().contentType(ContentType.JSON)
+                .when()
+                .body(Collections.singletonMap("approved", false))
+                .patch(String.format("/approvals/%s/firstLineApproval/%s?user=admin&group=managers", processId, taskId))
+                .then()
+                .statusCode(200)
+                .body("approved", is(false));
+    }
```
Comment (cristianonicolai) [inline integration-tests/integration-tests-springboot/src/it/integration-tests-springboot-it/src/test/java/org/kie/kogito/integrationtests/springboot/TaskTest.java:95]: maybe worth doing a check in the get method as well to ensure values were changed?

File: kogito-codegen-modules/kogito-codegen-processes/src/main/resources/class-templates/RestResourceUserTaskQuarkusTemplate.java [hunk 4]
```diff
@@ -7,6 +7,9 @@
 import org.kie.kogito.process.WorkItem;
 import org.kie.kogito.process.impl.Sig;
 
+
```
Comment (cristianonicolai) [inline kogito-codegen-modules/kogito-codegen-processes/src/main/resources/class-templates/RestResourceUserTaskQuarkusTemplate.java:10]: not needed? :)

File: integration-tests/integration-tests-springboot/src/it/integration-tests-springboot-it/src/test/java/org/kie/kogito/integrationtests/springboot/TaskTest.java (hunk 3 above)
Comment (cristianonicolai) [reply integration-tests/integration-tests-springboot/src/it/integration-tests-springboot-it/src/test/java/org/kie/kogito/integrationtests/springboot/TaskTest.java:95]: or perhaps add this call into the IT tests

Comment (fjtirado) [reply integration-tests/integration-tests-springboot/src/it/integration-tests-springboot-it/src/test/java/org/kie/kogito/integrationtests/springboot/TaskTest.java:95]: Reason Im not doing that is becuase @GET will return input paramenters, this affects output parameters.

**Comment by fjtirado** [comment]
@MarianMacik @cristianonicolai  @tiagodolphine 
finally green, please take a look and merge

**Review by cristianonicolai — APPROVED** [review]

**Comment by fjtirado** [comment]
@kiegroup/gatekeepers Can we merge this one

**Review by tiagodolphine — APPROVED** [review]

**Review by MarianMacik — APPROVED** [review]
Looks good to me.

File: api/kogito-api/src/main/java/org/kie/kogito/process/ProcessInstance.java [hunk 5]
```diff
@@ -110,6 +110,16 @@
...
+    
+    
+    /**
+     * Update work item output model without changing phase
+     *
+     * @param id        id of the work item to complete
+     * @param variables information that will replace existing output model
+     * @param policies  optional list of policies to be enforced
+     */
+    Map<String, Object> updateWorkItem(String id, Map<String, Object> variable, Policy<?>... policies);
```
[the remark this answers is not in the document]
Comment (fjtirado) [reply api/kogito-api/src/main/java/org/kie/kogito/process/ProcessInstance.java]: thanks done

**Comment by fjtirado** [comment]
Failed with /home/jenkins/workspace/KIE/kogito/pullrequest/kogito-runtimes/trace.sh: line 2: error.log: No space left on device, can be merged without problems

**Pull request merged**